packages/wallet/api/src/marshal-contexts.js

```diff
@@ -0,0 +1,257 @@
+// @ts-check
+import { makeScalarMap } from '@agoric/store';
+import { Far, makeMarshal, Remotable } from '@endo/marshal';
+import { HandledPromise } from '@endo/eventual-send'; // TODO: convince tsc this isn't needed
+
+const { details: X, quote: q } = assert;
+
+/**
+ * Make context for exporting wallet data where brands etc. can be recognized by boardId.
+ *
+ * Objects should be registered before serialization:
+ *   - for closely-held object, use initPurseId etc.
+ *   - for published objects such as brands and issuers, use initBoardId / ensureBoardId
+ *
+ * Unregistered objects (such as instances in amounts) get serialized
+ * with a fresh slot.
+ */
+export const makeExportContext = () => {
+  const myData = {
+    purse: {
+      /** @type {MapStore<string, Purse>} */
+      bySlot: makeScalarMap(),
+      /** @type {MapStore<Purse, string>} */
+      byVal: makeScalarMap(),
+    },
+    payment: {
+      /** @type {MapStore<string, Payment>} */
+      bySlot: makeScalarMap(),
+      /** @type {MapStore<Payment, string>} */
+      byVal: makeScalarMap(),
+    },
+    // TODO: offer, contact, dapp
+  };
+  const sharedData = {
```

Comment:
shared with whom? I think this is the case #2 in comment above so I suggest matching the language. e.g. `publishedData`.

is it all `boardData`?

Reply:
yes; it's all on the board. `Data` seems off too; you can't put a `CopyRecord` nor `CopyList` on the board; I think they have to be keys in the `@agoric/store` sense. Let's try `boardObjects`.